src/chain/chain.rs

```diff
@@ -324,33 +357,10 @@ impl Chain {
     /// Removes an elder from our section, creating a new `our_info` in the process.
     /// Should not be called while a pfx change is in progress.
     pub fn remove_elder(&mut self, pub_id: PublicId) -> Result<EldersInfo, RoutingError> {
-        if self.state.change != PrefixChange::None {
-            log_or_panic!(
-                LogLevel::Warn,
-                "Removing {:?} from chain during pfx change.",
-                pub_id
-            );
-        }
-
-        if !self.our_prefix().matches(&pub_id.name()) {
```

Comment:
Do we not want to log or panic here anymore ? I think it was put because it was entered in certain scenarios but OK to delete it if we are sure what that was and we are never going to hit this anymore ?

Reply:
The peer is already in out section and we wouldn't have added it if it didn't match our prefix in the first place. So this check would only fail if we tried to remove peer who is non a member of our section - but that case is already handled below.